src/clojure/nrepl/server.clj

```diff
@@ -133,11 +135,65 @@
   java.io.Closeable
   (close [this] (stop-server this)))
 
+
+;;; InetSockets (TCP)
+
+(defn inet-socket [bind port]
+  (let [port (or port 0)
+        addr (fn [^String bind ^Integer port] (InetSocketAddress. bind port))
+        ;; We fallback to 127.0.0.1 instead of to localhost to avoid
+        ;; a dependency on the order of ipv4 and ipv6 records for
+        ;; localhost in /etc/hosts
+        bind (or bind "127.0.0.1")]
+    (doto (ServerSocket.)
+      (.setReuseAddress true)
+      (.bind (addr bind port)))))
+
+
+;;; Unix domain (file namespace) sockets
+
+(def unix-socket-address-class
+  (memoize
+   (fn []
+     (try
+       (clojure.core/import* "org.newsclub.net.unix.AFUNIXSocketAddress")
+       (catch ClassNotFoundException ex
+         nil)))))
+
+(def unix-server-socket-class
```

Comment:
I seem that's almost the same as the previous function - I think creating a simple helper wouldn't hurt here.

Reply:
Added as `class-finder`.